### ISSUE TITLE
[Form] CollectionType entry_option "required => false" ignored in prototype since 2.8.1

### ISSUE THREAD

**Issue by bauer00** [symfony/symfony#17339]
I use the following code, where the Number-Field of 'scale' is required and the Number-Field of 'values' can be empty.
```
$builder
    ->add('scale', CollectionType::class, [
        'entry_type' => NumberType::class,
        'entry_options' => [
            'label' => false
        ],
        'allow_add' => true,
        'allow_delete' => true
    ])
    ->add('values', CollectionType::class, [
        'entry_type' => NumberType::class,
        'entry_options' => [
            'label' => false,
            'required' => false,
            'scale' => 0
        ],
        'allow_add' => true,
        'allow_delete' => true
    ]);
```

The generated HTML for existing fields is correct:
```
<input type="text" value="7" class="form-control" required="required" name="test[scale][6]" id="test_scale_6">
<input type="text" class="form-control" name="test[values][6]" id="test_values_6">
```

But the data-prototype is not, both fields are required:
```
<input type="text" id="test_scale___name__" name="test[scale][__name__]" required="required" class="form-control" />
<input type="text" id="test_values___name__" name="test[values][__name__]" required="required" class="form-control" />
```

I've found this issue: #15544
As far as I see, to fix this issue, the required-attribute of the child-field is overridden by the required-attribute of the CollectionType. But with "entry_options" since Symfony 2.8+ this causes this unexpectet behavior.
Of course it's possible for me to just define the required-attribute of the CollectionType, but for me if I set required = false on the CollectionType I expect the whole Collection is not required and can be null.

**Comment by HeahDude** [comment]
it's weird because the following test in 2.8 works fine :

```php
$form = $this->factory->create('Symfony\Component\Form\Extension\Core\Type\CollectionType', array(), array(
    'entry_type' => 'Symfony\Component\Form\Extension\Core\Type\NumberType',
    'allow_add' => true,
    'prototype' => true,
    'prototype_name' => '__test__',
    'entry_options' => [
        'required' => false
    ]
));

$this->assertTrue($form->createView()->vars['required'], 'collection should be required');
$this->assertFalse($form->createView()->vars['prototype']->vars['required'], '"prototype" should not be required');
```

**Comment by HeahDude** [comment]
I made a fork of symfony-strandard and used your scenario but could not reproduce your issue, it's working as expected

**Comment by bauer00** [comment]
I've installed a clean Symfony 2.8.1 project with the Symfony installer and tested your testcode with PHPUnit and it fails:
```
"prototype" should not be required
Failed asserting that true is false.
```
Revertet this project back to Symfony 2.8.0 and the test passes. Also tested it with clean 3.0.1 and 3.0.0, both passed.

**Comment by HeahDude** [comment]
@bauer00 alright, but test seems to pass on 2.8.2-dev, could you please confirm.
Also here's a fix for 2.8.1

Replace line 31 of `\Symfony\Component\Form\Extension\Core\CollectionType.php` by :
```php
'required' => isset($options['entry_options']['required']) ? $options['entry_options']['required'] : $options['required'],
```

**Comment by xabbuh** [comment]
This is a known regression introduced in Symfony 2.8.1 which was fixed in the meantime by #17162 and will be included in the 2.8.2 release.

**Comment by bauer00** [comment]
You're right, it's fixed in 2.8.2-dev. I didn't found #17162, thx for the hint.
